manubot/cite/tests/test_cite_command.py

```diff
@@ -65,38 +61,75 @@ def test_cite_command_file(tmpdir):
     not shutil.which('pandoc-citeproc'),
     reason='pandoc-citeproc installation not found on system'
 )
-def test_cite_command_render_stdout(args, expected):
-    """
-    Test the stdout output of `manubot cite --render` with various formats.
-    The output is sensitive to the version of Pandoc used, so rather than fail when
-    the system's pandoc is outdated, the test is skipped.
-    """
-    pandoc_version = get_pandoc_info()['pandoc version']
-    for output in 'markdown', 'html', 'jats':
-        if output in args and pandoc_version < (2, 5):
-            pytest.skip(f"Test {output} output assumes pandoc >= 2.5")
-    if pandoc_version < (2, 0):
-        pytest.skip("Test requires pandoc >= 2.0 to support --lua-filter and --csl=URL")
-    expected = (
-        pathlib.Path(__file__).parent
-        .joinpath('cite-command-rendered', expected)
-        .read_text()
-    )
-    args = [
-        'manubot', 'cite', '--render',
-        '--csl', 'https://github.com/greenelab/manubot-rootstock/raw/e83e51dcd89256403bb787c3d9a46e4ee8d04a9e/build/assets/style.csl',
-        'arxiv:1806.05726v1', 'doi:10.7717/peerj.338', 'pmid:29618526',
-    ] + args
-    process = subprocess.run(
-        args,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        universal_newlines=True,
-    )
-    print(shlex_join(process.args))
-    print(process.stdout)
-    print(process.stderr)
-    assert process.stdout == expected
+class Base_cite_command_render_stdout():
+    @classmethod
+    def expected_output(cls, filename):
+        return (
+            pathlib.Path(__file__).parent
+            .joinpath('cite-command-rendered', filename)
+            .read_text()
```

Comment:
let's specify `utf-8-sig` as per our convention

Reply:
Maybe need a more global helper funciton - I'll put a note on refactoring proposal in a comment.

Reply:
Apparently `utf-8-sig` https://ci.appveyor.com/project/manubot/manubot/builds/28002722/job/vy7a7kbigj665ewx causes problems on Appveyor, not sure why.

Reply:
```
    def test_no_arg(self):
>       assert self.render([]) == \
            self.expected_output('references-plain.txt')
E       AssertionError: assert '1. Generaliz... PMC5938574\n' == '1. Generaliz... PMC5938574\n'
E         Skipping 318 identical leading characters in diff, use -v to show
E         - peerj.338 · PMID: 24765572 · PMCID: PMC3994616
E         ?           -                 -
E         + peerj.338 � PMID: 24765572 � PMCID: PMC3994616
E           
E           3. Opportunities and obstacles for deep learning in biology and medicine.
E         - Travers Ching, Daniel S Himmelstein, Brett K Beaulieu-Jones, Alexandr A Kalinin, Brian T Do, Gregory P Way, Enrico Ferrero, Paul-Michael Agapow, Michael Zietz, Michael M Hoffman, … Casey S Greene
E         ?                                                                                                                                                                                    ^^^
E         + Travers Ching, Daniel S Himmelstein, Brett K Beaulieu-Jones, Alexandr A Kalinin, Brian T Do, Gregory P Way, Enrico Ferrero, Paul-Michael Agapow, Michael Zietz, Michael M Hoffman, � Casey S Greene
E         ?                                                                                                                                                                                    ^
E           Journal of the Royal Society, Interface (2018-04) https://www.ncbi.nlm.nih.gov/pubmed/29618526
E         - DOI: 10.1098/rsif.2017.0387 · PMID: 29618526 · PMCID: PMC5938574
E         ?                             -                 -
E         + DOI: 10.1098/rsif.2017.0387 � PMID: 29618526 � PMCID: PMC5938574
manubot\cite\tests\test_cite_command.py:106: AssertionError
```

Reply:
Which means system encoding is different on Appv and Travis machines? Strange, but let's stip the encoding with comment.

Reply:
> Which means system encoding is different on Appv and Travis machines?

Yeah, well AppVeyor is testing on Windows, Travis on Linux.

Reply:
> AppVeyor is testing on Windows

That I did not realise! The screens looked like Powershell though.